src/SFA.DAS.TrackProgress.Api/AppStart/AddConfiguration.cs

```diff
@@ -0,0 +1,60 @@
+﻿using SFA.DAS.Configuration.AzureTableStorage;
+
+namespace SFA.DAS.TrackProgress.Api.AppStart;
+
+public static class AddConfigurationExtensions
+{
+    public static WebApplicationBuilder AddConfiguration(this WebApplicationBuilder services)
+    {
+        AddAzureConfiguration(services.Configuration);
+
+#if DEBUG
+        services.Configuration.AddJsonFile($"appsettings.Development.json", optional: true);
+#endif
+
+        return services;
+    }
+
+    public static void AddAzureConfiguration(ConfigurationManager configuration)
+    {
+        if (configuration.IsLocalAcceptanceOrDev())
+            TryAddAzureConfiguration(configuration);
+        else
+            RequireAddAzureConfiguration(configuration);
+    }
+
+    public static void TryAddAzureConfiguration(ConfigurationManager configuration)
+    {
+        try
+        {
+            var names = configuration["ConfigNames"]?.Split(",") ?? Array.Empty<string>();
+            var connectionString = configuration["ConfigurationStorageConnectionString"];
+            var environment = configuration["EnvironmentName"];
+
+            if (names.Length == 0) return;
+            if (string.IsNullOrEmpty(connectionString)) return;
+
+            configuration.AddAzureTableStorage(options =>
+            {
+                options.ConfigurationKeys = names;
+                options.StorageConnectionString = connectionString;
+                options.EnvironmentName = environment;
+                options.PreFixConfigurationKeys = false;
+            });
+        }
+        catch
+        {
+        }
```

Comment:
We should really say why we can ignore the exception here. Is it because local storage may not be running? Also how does that affect the Unit tests when we run it as part of the build?

Reply:
Yes, a comment explaining why is a good idea.  It is indeed to cope with not having a local storage running.